### PR TITLE
Tiny: DRY default limit in ActiveRecord::Type::Integer

### PR DESCRIPTION
**Pull request by henrik**
Happened across this code and thought I'd name and DRY up the magic number 4.

### CODE REVIEW

File: activerecord/lib/active_record/type/integer.rb [hunk 1]
```diff
@@ -38,18 +38,21 @@ def cast_value(value)
...
-        1 << (limit * 8 - 1) # 8 bits per byte with one bit for sign
+        1 << (limit_or_default * 8 - 1) # 8 bits per byte with one bit for sign
       end
 
       def min_value
         -max_value
       end
+
+      def limit_or_default
+        limit || 4
```
Comment (vipulnsward) [inline activerecord/lib/active_record/type/integer.rb]: here magic number `4` can be extracted to a named variable/constant. Or the method can be better named to take care of that.

Comment (henrik) [inline activerecord/lib/active_record/type/integer.rb]: Yeah, I originally planned on just extracting a constant, but then I figured I could name it *and* DRY things up a little by extracting a method.

Any thoughts on a better method name, @vipulnsward?

As far as I can tell the "4" is for 4 byte storage size, which means a [MySQL `int`](https://ariejan.net/2009/08/20/once-and-for-all-rails-migrations-integer-limit-option/) or a [Postgres `integer`](http://www.postgresql.org/docs/9.4/static/datatype-numeric.html), as opposed to `smallint`, `bigint` etc. Guess a constant + a comment could capture some of that information.

**Comment by henrik** [comment]
@vipulnsward I force-pushed a new commit with a constant and a comment instead. Thoughts?

**Comment by senny** [comment]
@henrik looks good. Thank you.

**Comment by henrik** [comment]
@senny Thank you! :bouquet:

**Comment by vipulnsward** [comment]
:+1:

**Pull request merged**